modules/carto/src/api/common.ts

```diff
@@ -1,4 +1,22 @@
 export const DEFAULT_API_BASE_URL = 'https://gcp-us-east1.api.carto.com';
 export const DEFAULT_CLIENT = 'deck-gl-carto';
 export const V3_MINOR_VERSION = '3.4';
-export const MAX_GET_LENGTH = 8192;
+
+// Fastly default limit is 8192; leave some padding.
+export let MAX_GET_LENGTH = 7000;
+
+/**
+ * Returns maximum URL character length. Above this limit, requests use POST.
+ * Used to avoid browser and CDN limits.
+ */
+export function getRequestURLLimit(): number {
```

Comment:
I think it is cleaner to add `maxRequestLength` to `SourceOptionalOptions` and connect up in `fetchMap`, `baseSource` & `query`

Reply:
Done! ✅

Reply:
Could you also update the docs?

Reply:
Added to `data-sources.md`. I'd prefer not exposing the new parameter in pydeck-carto at this time, and haven't made changes there.